nreynoldson/beasties
# Configure db and built 'create-user' endpoint

- DynamoDB tables are configured
- API endpoint for creating a user - [https://idvmpyv72b.execute-api.us-east-1.amazonaws.com/dev/user](https://idvmpyv72b.execute-api.us-east-1.amazonaws.com/dev/user)
- API ID - idvmpyv72b
To test the endpoint with Postman: 
1. Select 'POST' request with request URL -  [https://idvmpyv72b.execute-api.us-east-1.amazonaws.com/dev/user](https://idvmpyv72b.execute-api.us-east-1.amazonaws.com/dev/user)
2. Authorization type - No Auth
3. Under Headers - Add Content-Type: application/json as key value pair
4. For Body, select Raw and add the following inside the text box
 {
    "userName": "aishwarya10397",
    "email": "aishwarya10397@gmail.com",
    "zipcode": "98087",
    "isShelterOwner": true,
    "shelterName": "shelter1"
}

Note: 
- For user table: userName, email and zipcode are required fields
- Also the formats of email should be correct (Eg: abc@abc.com) and zipcode (Eg: 00090)
- If isShelterOwner is true, shelterName is required
- I haven't given access to S3 bucket yet, so URL for avatar cannot be tested. It accepts any url, but in order to get the picture, we should be sending the image url from s3 bucket

Screenshots:
![Screen Shot 2022-02-04 at 9 06 18 PM](https://user-images.githubusercontent.com/52226049/152629561-4d28a4ae-af51-47f1-b85f-7ef105a8c70f.png)
